### PR TITLE
refactor(nuxt)!: cleanup schema and split nuxt 2 types

### LINKED ISSUE #5390: Detect typos in nuxt.config
Ref: https://github.com/nuxt/framework/issues/5294#issuecomment-1152108622_

We added support for `[key: string]: any]` in nuxt.config since 1) some modules might not have provided types and 2) making easier during module development to not have errors before stubbing

But this seems having a negative DX effect as increases chance of unnoticed typos in config file.

I think we shall try making it strict again...

### LINKED ISSUE #7753: More obvious deprecation of Nuxt 2 settings in `nuxt.config.js`
### Describe the feature

Hey 👋🏻

right now, it seems difficult for several people to realize which config options from `nuxt.config.js`/`nuxt.config.ts` are actually used in Nuxt 3 (e.g. `target`). Right now, the only hint is the `@version` tag in the JSDocs which isn't very prominent though.

image

It'd be nice to have a better alternative (e.g. `@deprecated`?) so people can be more aware.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

Resolves #5390, resolves #7753, resolves https://github.com/nuxt/framework/issues/8317

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Initially, we started Nuxt 3 project by documenting all possible configs from Nuxt 2 into `@nuxt/schema` alongside with `@nuxt/bridge` code. Over time it becomes both complex to maintain and also confusing UX because you can see auto-completion of options from `nuxt.config` that are not working with Nuxt 3.

Changes by this PR:

- Nuxt 2 legacy config types are splitted and moving out to nuxt/bridge (https://github.com/nuxt/bridge/pull/597)
- Hooks related to Nuxt 2 packages are splitted and moving out to nuxt/bridge (https://github.com/nuxt/bridge/pull/597)
- `@version` filter is removed from schema. Adding back for future schema versioning as `@nuxt` or `@constraints`

Breaking changes:

- Unknown and Nuxt2-only options are not allowed in `nuxt.config`
- `serverMiddleware` config is not supported anymore (use `serverHandlers` or `devServerHandlers`)
- `addServerMiddleware` kit utility is removed
- `webpack:devMiddleware` and `webpack:hotMiddleware` hooks unsupported
- `build:compile` and `build:compiled` hooks renamed to `webpack:compile` and `webpack:compiled`
- `build:resources` hook is unsupported (use `vite:compiled` and `webpack:compiled`)
- `build:before`, `build:done`, `modules:before` and `modules:done` take no arguments
- `bundler:change`, `bundler:error`, `bundler:done` and `bundler:progress` hooks renamed with `webpack:` prefix
- `options.env` and `options.target`, and `options.build.quite` are not used anymore by webpack builder

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/config/router.ts [hunk 1]
```diff
@@ -1,172 +1,18 @@
...
-   * @version 2
-   */
-  trailingSlash: undefined
+  router: {
+    /**
+     * Additional options passed to `vue-router`.
+     *
+     * Note: Only JSON serializable options should be passed by nuxt config.
+     *
+     * For more control, you can use `app/router.optionts.ts` file.
```
Comment (danielroe) [inline packages/schema/src/config/router.ts:10]:
```suggestion
     * For more control, you can use `app/router.options.ts` file.
```

File: packages/nuxt/src/core/nitro.ts [hunk 2]
```diff
@@ -185,61 +182,31 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
...
-
 declare module 'nitropack' {
   interface NitroRouteConfig {
     ssr?: boolean
   }
-  interface NitroRouteOptions {
+  interface NitroRouteRules {
     ssr?: boolean
   }
 }
```
Comment (danielroe) [inline packages/nuxt/src/core/nitro.ts:212]: I think we need to move this declaration to `packages/nuxt/types.d.ts`

Comment (pi0) [reply packages/nuxt/src/core/nitro.ts:212]: Yeah let's move it there (it makes issues with local tests only currently only)

Comment (danielroe) [reply packages/nuxt/src/core/nitro.ts:212]: It also causes issues in built app: https://github.com/nuxt/framework/issues/8317

**Comment by pi0** [comment]
(this PR has probably some breaking changes. merging earlier to find them out via edge channel)

**Pull request merged**